src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/magicnumber/InputMagicNumber_10.java

```diff
@@ -0,0 +1,118 @@
+        double double_magic1 = 1.5_0; // violation ''1.5_0' is a magic number'
+        if (1.0 < 2.0); // violation ''2.0' is a magic number'
+        double_magic1 *= 1.5; // violation ''1.5' is a magic number'
+    }
+}
+
+class C2 {
+    Person annon = new Person(55);
+    Object anon = new Object() {
+        public void magicMethod() {
+            Set<Person> p1 = new HashSet<Person>(5); // violation ''5' is a magic number'
+            Object[] values = new Object[3]; // violation ''3' is a magic number'
+            String[] fStrings = new String[3]; // violation ''3' is a magic number'
+            Person p12 = new Person(8); // violation ''8' is a magic number'
+
+            double double_var1 = 0D;
+            double_var1 = 1.0 + 2.0; // violation ''2.0' is a magic number'
+            double double_magic1 = 1.5_0; // violation ''1.5_0' is a magic number'
+            if (1.0 < 2.0); // violation ''2.0' is a magic number'
+            double_magic1 *= 1.5; // violation ''1.5' is a magic number'
+        }
+    };
+}
+
+class C3 {
+
+    C1 c = new C1(){
+        public void magicMethod() {
+            double double_var1 = 0D;
+            double_var1 = 1.0 + 2.0; // violation ''2.0' is a magic number'
+            double double_magic1 = 1.5_0; // violation ''1.5_0' is a magic number'
+            if (1.0 < 2.0); // violation ''2.0' is a magic number'
+            double_magic1 *= 1.5; // violation ''1.5' is a magic number'
+        }
+    };
+
+    final Object anon = new Object() {
+        public void magicMethod() {
+            double double_var1 = 0D;
+            double_var1 = 1.0 + 2.0; // violation ''2.0' is a magic number'
+            double double_magic1 = 1.5_0; // violation ''1.5_0' is a magic number'
+            if (1.0 < 2.0); // violation ''2.0' is a magic number'
+            double_magic1 *= 1.5; // violation ''1.5' is a magic number'
+        }
+    };
+}
+
+class C4 {
+    C2 cn = new C2(){
+        public void magicMethod() {
+            double double_var1 = 0D;
+            double_var1 = 1.0 + 2.0; // violation ''2.0' is a magic number'
+            double double_magic1 = 1.5_0; // violation ''1.5_0' is a magic number'
+            if (1.0 < 2.0); // violation ''2.0' is a magic number'
+            double_magic1 *= 1.5; // violation ''1.5' is a magic number'
+        }
+    };
+
+    final Object anon = new Object() {
+        public void magicMethod() {
+            double double_var1 = 0D;
+            double_var1 = 1.0 + 2.0; // violation ''2.0' is a magic number'
+            double double_magic1 = 1.5_0; // violation ''1.5_0' is a magic number'
+            if (1.0 < 2.0); // violation ''2.0' is a magic number'
+            double_magic1 *= 1.5; // violation ''1.5' is a magic number'
+        }
+    };
+}
+
+class Class1 {
+    Person person2 = new Person(12) { // violation ''12' is a magic number'
+        public void magicMethod() {
+            double double_var1 = 0D;
```

Comment:
I need your attention once again here.
The value 12 is used as an argument to the constructor of the Person class when a new Person object is created in the Class1 class. The Person constructor takes an int argument and initializes the Person object with that value.

A field, on the other hand, is a variable that is declared within a class and can hold values of various types, such as int, double, String, etc. Fields can be accessed by all the methods within the class. hence i think 12 is not a member of field

Reply:
```
kevin@inspiron-15-5510:~/Desktop/check_style$ cat Test2.java 
public class Test2 {
 
}

class Person {
        Person(int i) {}
}

class Class1 {

        Person p = new Person(13); // line no 11

    Person person = new Person(12) { // --> line no 13
        public void magicMethod() {
            double double_var1 = 0D;
            double_var1 = 1.0 + 2.0; 
       }
    };

}
kevin@inspiron-15-5510:~/Desktop/check_style$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="MagicNumber">

    </module>
  </module>
</module>
kevin@inspiron-15-5510:~/Desktop/check_style$ java -jar /home/kevin/Downloads/checkstyle-10.9.2-all.jar -c config.xml Test2.java
Starting audit...
[ERROR] /home/kevin/Desktop/check_style/Test2.java:11:31: '13' is a magic number. [MagicNumber]
[ERROR] /home/kevin/Desktop/check_style/Test2.java:13:32: '12' is a magic number. [MagicNumber]
Audit done.
Checkstyle ends with 2 errors.
kevin@inspiron-15-5510:~/Desktop/check_style$ 
```

Reply:
```
kevin@inspiron-15-5510:~/Desktop/check_style$ cat Test2.java 
public class Test2 {
 
}

class Person {
        Person(int i) {}
}

class Class1 {

        Person p = new Person(13); // line no 11

    Person person = new Person(12) { // --> line no 13
        public void magicMethod() {
            double double_var1 = 0D;
            double_var1 = 1.0 + 2.0; 
       }
    };

}kevin@inspiron-15-5510:~/Desktop/check_style$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="MagicNumber">
      <property name="ignoreFieldDeclaration" value="true" />
    </module>
  </module>
</module>kevin@inspiron-15-5510:~/Desktop/checjava -jar /home/kevin/Downloads/checkstyle-10.9.2-all.jar -c config.xml Test2.java
Starting audit...
Audit done.
kevin@inspiron-15-5510:~/Desktop/check_style$ 
```

Reply:
they are same

Reply:
I did not get it..
Why same version jar behaves different?

Reply:
I might miss something, please do single click output with execution of Released version. What we have with your changes we can see in Inputs.

Reply:
> I did not get it.. Why same version jar behaves different?

please look again both the configs are different.

My doubt is https://github.com/checkstyle/checkstyle/pull/12848#discussion_r1141596763 why this is not showing a violation? 
As per https://checkstyle.org/config_coding.html#MagicNumber:~:text=Ignore%20magic%20numbers%20in%20field%20declarations. it will only Ignore magic numbers in field declarations. and this is not a field variable as per my knowledge

Reply:
```
class Class1 {

        Person p
```

`p` is a field. User use code like `Color red = Color(123,124,53);` a lot, it is self explanatory, so extra variable is undesirable. Unfortunately other cases also hidden by this.

For code that your shared in this review thread, Check should work in same way after your update.

Reply:
done